osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -198,11 +205,32 @@ private void updateDisplay()
             updateDifficultyButtons();
         }
 
-        private void showBeatmap(IBeatmapInfo? beatmapInfo)
+        private void showBeatmap(APIBeatmap? beatmapInfo)
         {
+            guestMapperContainer.Clear();
+
+            if (beatmapInfo != null && beatmapSet?.Author.OnlineID != beatmapInfo.AuthorID)
+            {
+                if (BeatmapSet?.RelatedUsers?.Single(u => u.OnlineID == beatmapInfo.AuthorID) is APIUser user)
```

Comment:
This is a mess of conditionals. Also you are accessing both `beatmapSet` and `BeatmapSet` (should match other usages in file and only use one of the two).